core/com.b2international.snowowl.core/src/com/b2international/snowowl/core/request/resource/BaseTerminologyResourceCreateRequest.java

```diff
@@ -187,7 +208,52 @@ protected void preExecute(final TransactionContext context) {
 				.execute(context);
 		}
 	}
+	
+	@Override
+	@OverridingMethodsMustInvokeSuper
+	protected void checkParentCollection(TransactionContext context, Resource parentCollection) {
+		if (parentCollection instanceof TerminologyResourceCollection resourceCollection) {
+			// find the resource collection's tooling support for this child resource type
+			TerminologyResourceCollectionToolingSupport toolingSupport = context.service(TerminologyResourceCollectionToolingSupport.Registry.class).getToolingSupport(resourceCollection.getToolingId(), getResourceType());
+			
+			// append parent collection as a dependency of this new resource
+			final Dependency collectionDependency = Dependency.of(resourceCollection.getResourceURI());
+			setDependencies(dependencies == null ? List.of(collectionDependency) : ImmutableList.<Dependency>builder().addAll(dependencies).add(collectionDependency).build());
+			
+			// in case of matching child resource type, validate and inherit settings
+			inheritParentCollectionSettings(resourceCollection, toolingSupport.getInheritedSettingKeys());
+		} else if (parentCollection instanceof Bundle) {
+			// fetch all resources matching any of the current 
+			List<String> nonRootAncestorResourceIds = Collections3.toImmutableList(parentCollection.getResourcePathSegments()).stream().filter(path -> !IComponent.ROOT_ID.equals(path)).toList();
+			
+			if (!nonRootAncestorResourceIds.isEmpty()) {
+				// find and check the first matching terminology resource collection in the ancestor array (only one should exists)
+				ResourceRequests.prepareSearch()
+					.filterByResourceType(TerminologyResourceCollection.RESOURCE_TYPE)
+					.filterByIds(nonRootAncestorResourceIds)
+					.setLimit(nonRootAncestorResourceIds.size())
```

Comment:
According to the comment in line 230 this limit can be reduced to 2. Depending on the illegal-ness of such cases,  an exception can also be thrown if more than one result turns out to be a resource collection.

Reply:
Done